### PR TITLE
physic: flag.Value interface for various units.

### PR DESCRIPTION
**Pull request by NeuralSpaz**
Continuation of #303 adding more units that implement the flag.Value interface.
Add ElectricCurrent 
Add Pressure
Add Power
Add Energy
Add ElectricalCapacitance
Add LuminousIntensity
Add LuminousFlux

### CODE REVIEW

File: conn/physic/units.go [hunk 1]
```diff
@@ -524,6 +644,43 @@ func (e Energy) String() string {
 	return nanoAsString(int64(e)) + "J"
 }
 
+// Set sets the Energy to the value represented by s. Units are to
+// be provided in "Joule", "Joules" or "J" with an optional SI prefix: "p",
+// "n", "u", "µ", "m", "k", "M", "G" or "T".
+func (j *Energy) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: s/j/e/ ?

File: conn/physic/units.go [hunk 2]
```diff
@@ -545,6 +705,43 @@ func (c ElectricalCapacitance) String() string {
 	return picoAsString(int64(c)) + "F"
 }
 
+// Set sets the ElectricalCapacitance to the value represented by s. Units are
+// to be provided in "Farad", "Farads" or "F" with an optional SI prefix: "p",
+// "n", "u", "µ", "m", "k", "M", "G" or "T".
+func (f *ElectricalCapacitance) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: s/f/c/ ?

File: conn/physic/units.go [hunk 3]
```diff
@@ -573,6 +773,43 @@ func (l LuminousIntensity) String() string {
 	return nanoAsString(int64(l)) + "cd"
 }
 
+// Set sets the LuminousIntensity to the value represented by s. Units are to
+// be provided in "Candela", "Candelas" or "cd" with an optional SI prefix: "p",
+// "n", "u", "µ", "m", "k", "M", "G" or "T".
+func (c *LuminousIntensity) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: s/c/l/ ?

File: conn/physic/units.go [hunk 4]
```diff
@@ -599,6 +839,43 @@ func (f LuminousFlux) String() string {
 	return nanoAsString(int64(f)) + "lm"
 }
 
+// Set sets the LuminousFlux to the value represented by s. Units are to
+// be provided in "Lumen", "Lumens" or "lm" with an optional SI prefix: "p",
+// "n", "u", "µ", "m", "k", "M", "G" or "T".
+func (l *LuminousFlux) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: a/l/f/ ?

File: conn/physic/units.go [hunk 5]
```diff
@@ -503,6 +583,43 @@ func (p Power) String() string {
 	return nanoAsString(int64(p)) + "W"
 }
 
+// Set sets the Power to the value represented by s. Units are to
+// be provided in "Watt", "Watts" or "W" with an optional SI prefix: "p",
+// "n", "u", "µ", "m", "k", "M", "G" or "T".
+func (w *Power) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: s/w/p/ ?

File: conn/physic/units.go [hunk 6]
```diff
@@ -195,7 +235,7 @@ func (e ElectricResistance) String() string {
 // Set sets the ElectricResistance to the value represented by s. Units are to
 // be provided in "Ohm", "Ohms" or "Ω" with an optional SI prefix: "p", "n",
 // "u", "µ", "m", "k", "M", "G" or "T".
-func (f *ElectricResistance) Set(s string) error {
+func (r *ElectricResistance) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: String() uses `e` above. Please use the same one letter for every methods of a type. I failed to notice that in the previous change.

File: conn/physic/units.go [hunk 7]
```diff
@@ -111,6 +111,43 @@ func (e ElectricCurrent) String() string {
 	return nanoAsString(int64(e)) + "A"
```
Comment (maruel) [inline conn/physic/units.go]: this is an error it should have been `c`.

File: conn/physic/units.go [hunk 8]
```diff
@@ -111,6 +111,43 @@ func (e ElectricCurrent) String() string {
...
 }
 
+// Set sets the ElectricCurrent to the value represented by s. Units are to
+// be provided in "Amp", "Amps" or "A" with an optional SI prefix: "p", "n",
+// "u", "µ", "m", "k", "M", "G" or "T".
+func (a *ElectricCurrent) Set(s string) error {
```
Comment (maruel) [inline conn/physic/units.go]: s/a/c/

**Comment by NeuralSpaz** [comment]
I got stuck with speed.Set(s),  the rest needed to make sense.... ¯\_(ツ)_/¯ seemed like an not horrible idea to use base units at the time.

**Comment by maruel** [comment]
gohci

**Pull request merged**